build_ngtf.py

```diff
@@ -0,0 +1,400 @@
+    print( "OUTPUT WHL FILE: %s" % output_wheel)
+    
+    output_path = os.path.join(artifacts_location, output_wheel)
+    print( "OUTPUT WHL DST: %s" % output_path)
+    # Delete just in case it exists
+    try:
+        os.remove(output_path)
+    except OSError:
+        pass
+
+    # Now copy
+    shutil.copy2(output_wheel, artifacts_location)
+
+    os.chdir(pwd)
+    return output_wheel
+
+def install_ngraph_tf(venv_dir, ngtf_pip_whl):
+    # Load the virtual env
+    load_venv(venv_dir)
+
+    # Intall the ngtf_wheel
+    call(["pip", "install", "-U", ngtf_pip_whl])
+
+    import tensorflow as tf; 
+    print('TensorFlow version: r',tf.__version__); 
+    print(tf.__compiler_version__);
+    import ngraph_config; print(ngraph_config.__version__)
+
+    pass
+
+def download_repo(target_name, repo, version):
+
+    # First download to a temp folder
+    call(["git", "clone", repo, target_name])
+
+    # Next goto this folder nd determone the name of the root folder
+    pwd = os.getcwd()
+
+    # Go to the tree
+    os.chdir(target_name)
+
+    # checkout the specified branch
+    call(["git", "checkout", version])
+
+    os.chdir(pwd)
+
+def main():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '--debug_build',
+        help="Builds a debug version of the nGraph components\n",
+        action="store_true")
+
+    arguments = parser.parse_args()
+
+    if (arguments.debug_build):
+        print("Building in DEBUG mode\n")
+
+    #-------------------------------
+    # Recipe
+    #-------------------------------
+
+    # Create the build directory
+    # mkdir build directory
+    build_dir = 'build'
+    try:
+        os.makedirs(build_dir)
+    except OSError as exc:  # Python >2.5
+        if exc.errno == errno.EEXIST and os.path.isdir(build_dir):
+            pass
+
+    pwd = os.getcwd()
+    os.chdir(build_dir)
+
+    # Component versions
+    ngraph_version = "6e06cded1d30030136d5677e0b3851dd4cc04bee"
+    tf_version = "v1.12.0"
+
+    # Download TensorFlow
+    download_repo(
+        "tensorflow", 
+        "https://github.com/tensorflow/tensorflow.git", 
+        tf_version)
+    # Build TensorFlow
+    build_tensorflow("venv-tf-py3", "tensorflow", "artifacts")
+
+    # Install tensorflow
+    cxx_abi = install_tensorflow("venv-tf-py3", "artifacts")
+
+    # Download nGraph
+    download_repo(
+        "ngraph", 
+        "git@github.com:NervanaSystems/ngraph.git", 
+        ngraph_version)
+
+    # Now build nGraph
+    artifacts_location = os.path.abspath("artifacts")
+    print("ARTIFACTS location: " + artifacts_location)
+    
+    ngraph_cmake_flags = [
+        "-DNGRAPH_INSTALL_PREFIX=" + artifacts_location,
+        "-DNGRAPH_DISTRIBUTED_ENABLE=YES", "-DNGRAPH_USE_CXX_ABI=" + cxx_abi,
+        "-DNGRAPH_UNIT_TEST_ENABLE=NO", "-DNGRAPH_TOOLS_ENABLE=YES",
+        "-DNGRAPH_DEX_ONLY=TRUE", "-DNGRAPH_GPU_ENABLE=NO",
+        "-DNGRAPH_PLAIDML_ENABLE=NO", "-DNGRAPH_DEBUG_ENABLE=NO"
```

Comment:
`-DNGRAPH_UNIT_TEST_ENABLE=NO"`->So this script currently builds without unit tests.

We could pass these along as options to the script (for example we already have arguments.debug_build)
or we could leave it for a later PR. But in the readme/doc/helpline of the script we can add exactly what configuration this script currently supports (eg, ngtf, not tfng, unit test not built, python 3, no mkl) etc

Another possibility is to have a json file of compile configs instead of `argparse.ArgumentParser`